products/spanner/terraform.yaml

```
@@ -63,7 +63,7 @@ overrides: !ruby/object:Overrides::ResourceOverrides
 
           If not provided, a random string starting with `tf-` will be selected.
         validation: !ruby/object:Provider::Terraform::Validation
-          regex: '^(?:[a-z](?:[-_a-z0-9]{4,28}[a-z0-9])?)$'
+          regex: '^(?:[a-z](?:[-_a-z0-9]{4,62}[a-z0-9])?)$'
```

```suggestion
          regex: '^([a-z]([-_a-z0-9]{0,62}[a-z0-9]))$'
```

I lied, I played with it more and I think this does it. Notably remove `?` which makes the entire second group optional, and make it so that a range from 2 - 64 is allowed.

I am much more convinced we should just remove this regex rather than improve it.

---

I have NO IDEA where that 4 came from or why I didn't see it